database-generic/src/Database/Generic/Class.hs

```diff
@@ -9,7 +9,7 @@ import Database.Generic.Statement.Output (HasOutputType, Output, OutputError, Ou
 
 -- | Monads that can communicate with a database over a given connection.
 class (Exception (Error m t), Functor t, Monad m, Show (Error m t))
-  => MonadDb m t c | m -> c where
+  => MonadDb m t c dbv | m -> c, m -> dbv where
```

Comment:
Avoid new fundep. Consider a new type parameter to decide which DB to connect to